remote/github/github.go

```diff
@@ -330,3 +346,162 @@ func (g *Github) GetHook(r *http.Request) (*model.Hook, error) {
+
+	log.Debug(*hook)
+
+	return hook, nil
+}
+
+func (g *Github) GetPullRequestsForCommit(u *model.User, r *model.Repo, sha *string) ([]model.PullRequest, error) {
+	client := setupClient(g.API, u.Token)
+	fmt.Println("sha == ", sha, *sha)
+	issues, _, err := client.Search.Issues(fmt.Sprintf("%s&type=pr", *sha), &github.SearchOptions {
+		TextMatch: false,
+	})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]model.PullRequest, len(issues.Issues))
+	for k, v := range issues.Issues {
+		pr, _, err := client.PullRequests.Get(r.Owner, r.Name, *v.Number)
+		if err != nil {
+			return nil, err
+		}
+
+		mergeable:= true
+		if pr.Mergeable != nil {
+			mergeable = *pr.Mergeable
+		}
+
+		status, _, err := client.Repositories.GetCombinedStatus(r.Owner, r.Name, *sha, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		out[k] = model.PullRequest{
+			Issue: model.Issue{
+				Number: *v.Number,
+				Title: *v.Title,
+				Author: *v.User.Login,
+			},
+			Branch:  model.Branch {
+				Name: *pr.Head.Ref,
+				BranchStatus: *status.State,
+				Mergeable: mergeable,
+
+			},
+		}
+	}
+	return out, nil
+}
+
+func (g *Github) GetBranchStatus(u *model.User, r *model.Repo, branch string) (*model.BranchStatus, error) {
+	client := setupClient(g.API, u.Token)
+	statuses, _, err := client.Repositories.GetCombinedStatus(r.Owner, r.Name, branch, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return (*model.BranchStatus)(statuses.State), nil
+}
+
+func (g *Github) MergePR(u *model.User, r *model.Repo, pullRequest model.PullRequest) (*string, error) {
+	client := setupClient(g.API, u.Token)
+
+	result, _, err := client.PullRequests.Merge(r.Owner, r.Name, pullRequest.Number, "Merged by LGTM")
+	if err != nil {
+		return nil, err
+	}
+
+	if !(*result.Merged) {
+		return nil, errors.New(*result.Message)
+	}
+	return result.SHA, nil
+}
+
+func (g *Github) GetMaxExistingTag(u *model.User, r *model.Repo) (*version.Version, error) {
+	client := setupClient(g.API, u.Token)
+
+	//find the previous largest semver value
+	var maxVer *version.Version
+
+	tags, _, err := client.Repositories.ListTags(r.Owner, r.Name, nil)
+	if err != nil {
+		return nil, err
+	}
+	for _, v := range tags {
+		curVer, err := version.NewVersion(*v.Name)
+		if err != nil {
+			continue
+		}
+		if maxVer == nil || curVer.GreaterThan(maxVer) {
+			maxVer = curVer
+		}
+	}
+
+	if maxVer == nil {
+		maxVer, _ = version.NewVersion("v0.0.0")
+	}
+	log.Debugf("maxVer found is %s", maxVer.String())
+	return maxVer, nil
+}
+
+func (g *Github) Tag(u *model.User, r *model.Repo, version *version.Version, sha *string) error {
+	client := setupClient(g.API, u.Token)
+
+	t := time.Now()
+	tag, _, err := client.Git.CreateTag(r.Owner, r.Name, &github.Tag{
+		Tag:     github.String(version.String()),
+		SHA:     sha,
+		Message: github.String("Tagged by LGTM"),
+		Tagger: &github.CommitAuthor{
+			Date:  &t,
+			Name:  github.String("LGTM"),
+			Email: github.String("LGTM@lgtm.co"),
```

Comment:
also maybe make this configurable as part of the github configuration? Not sure if people will care / want to customize or not. Open to suggestions here. If not, we can ignore the idea